tests/lib/rules/jsx-sort-props.js

```diff
@@ -494,6 +608,214 @@ ruleTester.run('jsx-sort-props', rule, {
       output: '<App z onBar />;',
       options: reservedFirstAndCallbacksLastArgs,
       errors: [expectedCallbackError],
+    // multiline first
+    },
+    {
+      code: `
+        <App
+          a
+          b={{
+            bB: 1,
+          }}
+        />
+      `,
+      options: multilineFirstArgs,
+      errors: [expectedMultilineFirstError],
+      output: `
+        <App
+          b={{
+            bB: 1,
+          }}
+          a
+        />
+      `,
+    },
+    {
+      code: `
+        <App
+          a={1}
+          b={{
+            bB: 1,
+          }}
+          c
+        />
+      `,
+      options: multilineAndShorthandFirstArgs,
+      errors: [expectedMultilineFirstError, expectedShorthandFirstError],
+      output: `
+        <App
+          c
+          b={{
+            bB: 1,
+          }}
+          a={1}
+        />
+      `,
+    },
+    // multiline last
+    {
+      code: `
+        <App
+          a={{
+            aA: 1,
+          }}
+          b
+        />
+      `,
+      options: multilineLastArgs,
+      errors: [expectedMultilineLastError],
+      output: `
+        <App
+          b
+          a={{
+            aA: 1,
+          }}
+        />
+      `,
+    },
+    {
+      code: `
+        <App
+          a={{
+            aA: 1,
+          }}
+          b
+          inline={1}
+          onClick={() => ({
+            c: 1
+          })}
+          d="dD"
+          e={() => ({
+            eE: 1
+          })}
+          f
+        />
+      `,
+      options: multilineAndShorthandAndCallbackLastArgs,
+      errors: [
+        {
+          messageId: 'listShorthandLast',
+          line: 6,
+        },
+        {
+          messageId: 'listCallbacksLast',
+          line: 8,
+        },
+        {
+          messageId: 'listCallbacksLast',
+          line: 8,
+        },
+        {
+          messageId: 'listCallbacksLast',
+          line: 8,
+        },
```

Comment:
three reports on the same line seems like an issue; we may need to do some memoizing (via a WeakMap) to prevent reporting on a node we've already reported on.
```suggestion
        {
          messageId: 'listShorthandLast',
          line: 6,
        },
        {
          messageId: 'listCallbacksLast',
          line: 8,
        },
```

Reply:
Makes sense, I just added a method to memoize the reported node attributes